jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesPresenter.java

```diff
@@ -48,120 +58,164 @@ public String getTittle() {
-        return cds.getRoles().keySet().stream().filter(
-                role -> {
-                    if ("owner".equals(role)) {
-                        return false;
-                    }
-                    final Integer roleCardinality = cds.getRoles().get(role);
-                    if (roleCardinality == -1) {
-                        return true;
-                    }
-                    final Integer roleInstanceCardinality = cis.getRoleAssignments().stream().filter(ra -> role.equals(ra.getName())).findFirst().map(ra -> ra.getGroups().size() + ra.getUsers().size()).orElse(0);
-                    return roleInstanceCardinality < roleCardinality;
+    private CaseRoleAction getEditAction(CaseRoleAssignmentSummary caseRoleAssignmentSummary,
+                                         boolean hasAssignment, boolean enabledAction) {
+        return new CaseRoleAction() {
+            @Override
+            public String label() {
+                if (hasAssignment) {
+                    return translationService.format(EDIT);
                 }
-        ).collect(Collectors.toSet());
-    }
+                return translationService.format(ASSIGN);
+            }
 
-    protected void setupRoleAssignments(final CaseInstanceSummary cis) {
-        if (cis.getRoleAssignments() == null || cis.getRoleAssignments().isEmpty()) {
-            return;
-        }
+            @Override
+            public boolean isEnabled() {
+                return enabledAction;
+            }
 
-        cis.getRoleAssignments().forEach(
-                crs -> {
-                    crs.getUsers().forEach(user -> view.addUser(user, crs.getName(), new CaseRoleAction() {
-
-                        @Override
-                        public String label() {
-                            return translationService.format(REMOVE);
-                        }
-
-                        @Override
-                        public void execute() {
-                            removeUserFromRole(user, crs.getName());
-                        }
-                    }));
-                    crs.getGroups().forEach(group -> view.addGroup(group, crs.getName(), new CaseRoleAction() {
-
-                        @Override
-                        public String label() {
-                            return translationService.format(REMOVE);
-                        }
-
-                        @Override
-                        public void execute() {
-                            removeGroupFromRole(group, crs.getName());
-                        }
-                    }));
-                }
-        );
+            @Override
+            public void execute() {
+                editRoleAssignmentView.show(caseRoleAssignmentSummary,
+                        () -> assignToRole(caseRoleAssignmentSummary.getName(),
+                                editRoleAssignmentView.getUsersNames(),
+                                editRoleAssignmentView.getGroupsNames()
```

Comment:
Model (caseRoleAssignmentSummary) is already present as the first argument of in the _show_ call.
Why not calling directly:
caseRoleAssignmentSummary.getUsers() and 
caseRoleAssignmentSummary.getGroups()
instead of: 
editRoleAssignmentView.getUsersNames()
editRoleAssignmentView.getGroupsNames()?

Taking this into account, getUsersNames() and getGroupsNames() are unnecessary in EditRoleAssignmentView interface.